strawberry_django_plus/middlewares/user_warmup.py

```diff
@@ -0,0 +1,27 @@
+import inspect
+from typing import Any, Awaitable, Callable, Coroutine, cast
+
+from asgiref.sync import sync_to_async
+from django.http import HttpRequest, HttpResponse
+from django.utils.decorators import sync_and_async_middleware
+
+
+@sync_and_async_middleware
+def user_warmup_middleware(
+    get_response: Callable[[HttpRequest], HttpResponse | Coroutine[Any, Any, HttpResponse]],
```

Comment:
You probably need to change the `|` by `Union`, because this lib targets python 3.8+

Or you can `from __future__ import annotations` at the top of the file

Reply:
Jupp, added those changes in a clean up typing commit. Pyright should be good